### PR TITLE
feat(aws-s3-stepfunctions): Changed escape hatch to eventBridgeEnabled prop

### LINKED ISSUE #565: feat(aws-s3-stepfunctions): Replace EventBridge enabled property override with prop
The S3 cfnBucket class has created new prop called [eventBridgeEnabled](https://github.com/aws/aws-cdk/issues/18076) in S3 bucket props. We will replace the override code line 

```cfnBucket.addPropertyOverride('NotificationConfiguration.EventBridgeConfiguration.EventBridgeEnabled', true);```

with the newly added prop to properly create new S3 buckets with EventBridge enabled. 

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by mickychetta**
*Issue #565 , if available:*

*Description of changes:*
- Changed escape hatch to eventBridgeEnabled prop
- Generated new integ expected JSON
- Updated unit tests for Custom EventBridge Enabled 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

fixes #565

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 8023dbe91b60adc4b9ae8e8c5ebacff003fb3992
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3f2975ed-0bfa-4c1d-8b93-8bd85a35c757%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: 8023dbe91b60adc4b9ae8e8c5ebacff003fb3992
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=dc4d1e73-a97d-4e01-afbb-a3340082f3fa%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 487ea23c9857f007372762e42ac82b99738744f2
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a7df2db4-46e2-4389-85fd-30d4afd02521%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: 487ea23c9857f007372762e42ac82b99738744f2
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3fb5a02a-5625-489a-8997-598eebfaf04c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: b05dc8cff8d979b1bad29e9da8affaf7b438af2b
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=074ad8dc-aa2a-47e0-871b-90e69e92f8d8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: b05dc8cff8d979b1bad29e9da8affaf7b438af2b
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9ba433da-f1ce-460f-a5e5-23b782b82e8e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: d82f1945bbbb9746d3dd9c8b5c61d78801d0761f
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7d36564e-fa82-418f-84ac-0251d92f8641%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: d82f1945bbbb9746d3dd9c8b5c61d78801d0761f
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bcd24235-3175-4f7e-8976-9ee28ae4e8d1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-s3-step-function/test/integ.pre-existing-bucket.ts [hunk 1]
```diff
@@ -15,16 +15,15 @@
...
+import { generateIntegStackName, addCfnNagS3BucketNotificationRulesToSuppress } from '@aws-solutions-constructs/core';
 
 const app = new App();
 const stack = new Stack(app, generateIntegStackName(__filename));
 
-const existingBucket = CreateScrapBucket(stack, {});
-const cfnBucket = existingBucket.node.defaultChild as s3.CfnBucket;
-cfnBucket.addPropertyOverride('NotificationConfiguration.EventBridgeConfiguration.EventBridgeEnabled', true);
+const existingBucket = CreateScrapBucket(stack, {
+  eventBridgeEnabled: true
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-step-function/test/integ.pre-existing-bucket.ts:25]: Is there now a requirement that if an existing bucket is provided that eventBridgeEnabled is set to true? We should document that in the README. Better yet, can we turn it on ourselves? (I'm guessing that if this accepts an IBucket then we are unable to do so...)

Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-step-function/test/integ.pre-existing-bucket.ts:25]: Is there now a requirement that if an existing bucket is provided that eventBridgeEnabled is set to true? We should document that in the README. Better yet, can we turn it on ourselves? (I'm guessing that if this accepts an IBucket then we are unable to do so...)

File: source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/lib/index.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export interface S3ToStepfunctionsProps {
   /**
    * Optional user provided props to override the default props for the S3 Bucket.
    *
-   * @default - Default props are used
+   * @default - Default props are usedÎ
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/lib/index.ts:37]: Typo? (extra weird character)

File: source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/lib/index.ts [hunk 3]
```diff
@@ -34,7 +34,7 @@ export interface S3ToStepfunctionsProps {
   /**
    * Optional user provided props to override the default props for the S3 Bucket.
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/lib/index.ts:35]: Here's where we need a comment that the events feature must already be turned on for an existing bucket as this construct can't do it from an IBucket (I assume...)

And in the README.md

File: source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/lib/index.ts [hunk 4]
```diff
@@ -110,15 +110,14 @@ export class S3ToStepfunctions extends Construct {
 
     if (!props.existingBucketObj) {
       [this.s3Bucket, this.s3LoggingBucket] = defaults.buildS3Bucket(this, {
-        bucketProps: props.bucketProps,
+        bucketProps: defaults.consolidateProps({ eventBridgeEnabled: true }, props.bucketProps),
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-s3-stepfunctions/lib/index.ts:113]: This is backwards, you've got the key prop for the construct as a default prop which can be overwritten. You need to make it the third argument, constructProps, which are props that the construct MUST turn on, so overwrite any customer specification.

defaults.consolidateProps({}, props.bucketProps, { eventBridgeEnabled: true });

**Review by biffgaut — CHANGES_REQUESTED** [review]

File: source/patterns/@aws-solutions-constructs/aws-s3-step-function/test/integ.pre-existing-bucket.ts (hunk 1 above)
Comment (mickychetta) [reply source/patterns/@aws-solutions-constructs/aws-s3-step-function/test/integ.pre-existing-bucket.ts:25]: Correct, we are unable to turn it on ourselves

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: a3ed44c4b2365c9c163db26d00777e13c12514ac
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f67b176d-a577-4f54-9d1b-9e663d46f6db%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: a3ed44c4b2365c9c163db26d00777e13c12514ac
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ebbbe726-4dd9-4bb8-8d7e-e81fe9830aef%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Pull request merged**